### DISCUSSION TITLE
VivadoAccelerator for U250 card

### DISCUSSION THREAD

**Discussion by Sarika124** [fastmachinelearning/hls4ml#671] [Q&A] [answered]
Hi..
Is _VivadoAccelarator_ of hls4ml will support U250 card? If we want to deploy cnn model to u250 card what all the changes we need to do accordingly on the code??
Any leads would be appreciated.

**Comment by thesps** [comment, marked as answer]
I transferred the issue to hls4ml. In the current main branch hls4ml does support U250 cards. I recommend this way of getting the initial VivadoAccelerator configuration template from the backend, specifying the board. Then you need to specify the same 'Platform' as the one you have installed on your system. I have an Alveo U200, with `xilinx_u200_xdma_201830_2` platform, so my code looks like:
```
config = hls4ml.backends.get_backend('VivadoAccelerator').create_initial_config(board='alveo-u200')
config['AcceleratorConfig']['Platform'] = 'xilinx_u200_xdma_201830_2'
config['HLSConfig'] = hls_config
config['KerasModel'] = model
config['OutputDir'] = '...'
config['ProjectName'] = '...'
config['Backend'] = 'VivadoAccelerator'
hls_model = hls4ml.converters.keras_to_hls(config)
  ```

And note that currently this is only working with Vivado 2020.1

**Reply by jmduarte** [reply]
@Sarika124 I migrated this to a Discussion. If @thesps's answer works for you, please mark it as the answer! Thank you!

**Reply by deleteeeee** [reply]
helllo, does VivadoAccelarator of hls4ml support U280?

**Reply by vloncar** [reply]
@deleteeeee according to [this](https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/vivado_accelerator/supported_boards.json#L36) line, yes.
